Change revision queries to use the new actor table/columns

dayllanmaza opened the pull request:
I'm not  sure we still need to use the revision_userindex but https://wikitech.wikimedia.org/wiki/Help:Toolforge/Database has not been updated to reflect the actors table or if there is any other optimized table view with the new revision/actor/user relationship.

This should do it for now.

Bug: T224613

dayllanmaza committed:
Fix revision queries with actor columns

Bug: T224613

MusikAnimal reviewed (approved):
No other views for `actor`, and yes continue to `revision_userindex`. What you have here looks correct, though I have not tested it.

davidbarratt commented:
It looks like you need to update the travis config to use PHP 7.2

dayllanmaza committed:
Fix test and update travis ci to use php 7.2

dayllanmaza commented:
@MusikAnimal thank you.
@davidbarratt It was a test failing due to the change in column names. I fixed that and updated Travis anyway to match the docker image.
